src/main/java/com/ebivariation/contigalias/controller/authentication/SecurityConfiguration.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ebivariation.contigalias.controller.authentication;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+    private static final String REALM = "EBI-REALM";
+
+    private static final String ROLE_ADMIN = "ADMIN";
+
+    private static final String USERNAME_ADMIN = "contig-admin";
+
+    private static final String USERNAME_PASSWORD = "Souper$ecurE123";
```

Comment:
The problem with this approach is that usernames and passwords are hard coded as clear text. Should I try to move them to a separate configuration file that won't be included in the GitHub repository?

Reply:
we can do this with maven profiles. I will do the changes for you because it's a bit obscure and it's not documented very well.

Reply:
I have already moved these to application.properties in a commit following this comment.